datumaro/plugins/data_formats/datumaro/base.py

```diff
@@ -53,6 +68,29 @@ def __init__(self, path):
         super().__init__(subset=osp.splitext(osp.basename(path))[0])
         self._load_impl(path)
 
+    @classmethod
+    def get_dm_version(cls, path: str):
+        """
+        Get Datumaro library version at exporting the dataset
+
+        Note that the regacy Datumaro doesn't store the version into exported dataset.
+        Thus it returns DatumaroBase.REGACY_VERSION when
+        """
+        versions = []
+        for annot_file in glob.glob(
+            osp.join(path, DatumaroPath.ANNOTATIONS_DIR, "**", "*" + DatumaroPath.ANNOTATION_EXT),
+            recursive=True,
+        ):
+            versions.append(parse_json_file(annot_file).get("dm_version", cls.REGACY_VERSION))
```

Comment:
This will make JSON parsing twice: 1) version check and 2) dataset content parsing. In the latest experiment I did, JSON parsing the Datumaro format of COCO instance segmentation takes ~10 seconds. This is non-negligible overhead. We need to make it do just once.

Reply:
Yes, exactly.

I let a DatumaroBase instance stores the parsed_json at the init() to avoid duplicated parsing.